### PR TITLE
Use Java 8 map.putIfAbsent and map.computeIfAbsent #8215

### LINKED ISSUE #8215: Java: Use Java 8 map.putIfAbsent and map.computeIfAbsent
Context: Someone wants to add a key-value pair to a map if the key-value pair does not exist.

Pre-Java 8:
```java
if (!map.containsKey(key)) {
    map.put(key, value);
}
```
Java 8:
```java
// If value is independent of key
map.putIfAbsent(key, value);

// If value is dependent of key
map.computeIfAbsent(key, key -> process(key));
```

For this one there are more than 20 places. Have to fix no less than 20 to qualify for a valid PR.

### PR DESCRIPTION
**Pull request by pedrolarben**
Part of #8215 
_Ready for review_

I have change some methods in order to use the java8 methods: map.putIfAbsent, map.computeIfAbsent and map.computeIfPresent. 

I have changed all the possible methods that I have found, except for those that have already being fixed in the ongoing PR #8705

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/InstructorPrivileges.java [hunk 1]
```diff
@@ -269,10 +269,10 @@ private void updatePrivilegesInSessionLevel(String sectionName, String sessionNa
     }
 
     private void verifyExistenceOfsectionName(String sectionName) {
-        if (!this.sessionLevel.containsKey(sectionName)) {
+        this.sessionLevel.computeIfAbsent(sectionName, key -> {
             addSectionWithDefaultPrivileges(sectionName);
-            this.sessionLevel.put(sectionName, new LinkedHashMap<String, Map<String, Boolean>>());
-        }
+            return new LinkedHashMap<String, Map<String, Boolean>>();
```
Comment (wenmogu) [inline src/main/java/teammates/common/datatransfer/InstructorPrivileges.java]: you can simply return `new LinkedHashMap<>`

Comment (pedrolarben) [reply src/main/java/teammates/common/datatransfer/InstructorPrivileges.java]: I decided to return `new LinkedHashMap<String, Map<String, Boolean>>` insted of simply `new LinkedHashMap` in order to try to not change the old code. In the old code we were putting a `new LinkedhashMap<String, Map<String, Boolean>>` in the map `this.sessionLevel` so I decided to do the same.


**Old Code**
```java
    private void verifyExistenceOfsectionName(String sectionName) {
        if (!this.sessionLevel.containsKey(sectionName)) {
            addSectionWithDefaultPrivileges(sectionName);
            this.sessionLevel.put(sectionName, new LinkedHashMap<String, Map<String, Boolean>>());
        }
    }
```
**New Code**
```java
    private void verifyExistenceOfsectionName(String sectionName) {
        this.sessionLevel.computeIfAbsent(sectionName, key -> {
            addSectionWithDefaultPrivileges(sectionName);
            return new LinkedHashMap<String, Map<String, Boolean>>();
        });
    }
```

Comment (wenmogu) [reply src/main/java/teammates/common/datatransfer/InstructorPrivileges.java]: Yeah I know, but that part is unnecessary even in the old code. Not all parts of the old code e correct :P

Comment (wenmogu) [reply src/main/java/teammates/common/datatransfer/InstructorPrivileges.java]: Hi @pedrolarben you r right, they r equivalent. We shall change it to make the code cleaner :)

**Comment by pedrolarben** [comment]
Changes made

**Comment by pedrolarben** [comment]
@wenmogu I'm having problem executing the npm lint test, should I open a new issue to ask for help?
`Error: Cannot find module 'webpack-merge'`

**Comment by pedrolarben** [comment]
@wenmogu  I have changed the code. I have fixed the code style


**Tests**
- [X] Code Style test  
- [X] Local tests 
- [ ] CI tests
 
I need help with the CI tests because there are some tests failing :(

**Comment by wenmogu** [comment]
oh? as long as you can pass the Travis CI test online here :)
I know there are some random tests failing e.g. some EditStudentUITest or RankQuestionUITest... try run the failed tests a few times generally it will pass eventually (if you didnt modify any relevant code)... i guess its due to some time lag that selenium couldnt handle at times...

**Comment by wenmogu** [comment]
hi @wkurniawan07 can pls take a look at this? shld be a fast read :D 

I think after this PR and the other ongoing PR #8705  this issue can be officially closed. I guess another first timer issue about replacement using `computeIfPresent` can be done as there are quite some places (arnd 10) that can be shortened using this method.

**Comment by pedrolarben** [comment]
Change done @wkurniawan07

**Pull request merged**